docs/get-started/3d-mesh/index.md

```diff
@@ -0,0 +1,303 @@
+the mentioned label coming with every `Document` which we specified during preprocessing and will sample it as a positive example
+and then sample examples with different label as negatives.
+
+The triplet loss's goal is to pull `Documents` with same class together and those with different classes away from each other thus improving
+the final representation.
+
+```python
+tuned_model = finetuner.fit(
+    model,
+    train,
+    eval_data=eval,
+    preprocess_fn=partial(preprocess, num_points=2048, data_aug=True),
+    epochs=50,
+    batch_size=64,
+    loss=TripletLoss(
+        miner=TripletEasyHardMiner(pos_strategy='easy', neg_strategy='semihard')
+    ),
+    configure_optimizer=configure_optimizer,
+    learning_rate=5e-4,
+    device='cuda',
+)
+# saving the finetuner model
+torch.save(
+    tuned_model.state_dict(),
+    str(checkpoint_dir / f'finetuned-{model_name}-d{embed_dim}.pth'),
+)
+```
+
+## Evaluating embedding quality & Results
+
+Now let's see whether we made an improvement or not. We will use the training data as index and the testing data as query. We'll embed them both
+and then search for test 3d objects (the query) in our training data (index). We do this using the pretrained model and the finetuned model and then compare top matches.
+
+We'll use two metrics to evaluate our two models (pretrained, straight out of the box model and finetuned model) on the search task:
+
+**mAP@k** : We'll calculate the average precision at 1, 5 and 10, then we'll calculate the mean of those average precisions for all documents in our test data (these are our queries) because we care about how accurate and precise our retrieved 3D objects are.
+
+**mNDCG@k**: We'll calculate NDCG at 1,5 and 10, then we'll calculate the mean of those for all documents in our test data  because we care about the order of the retrieved 3D objects.
+
+````{dropdown} Complete source code
+
+```python
+from finetuner.tuner.evaluation import Evaluator
+
+train = DocumentArray.load_binary('../train_data_modelnet.bin') # load train dataset
+eval = DocumentArray.load_binary('../test_data_modelnet.bin') # load test dataset
+
+train.apply(preprocess)
+eval.apply(preprocess)
+
+tuned_model = MeshDataModel(model_name='pointconv', embed_dim=512) # create pointconv model with 512 dimensions
+pretrained_model = MeshDataModel(model_name='pointconv', embed_dim=512)
+
+tuned_model.load_state_dict(torch.load('checkpoints/finetuned-pointconv-d512.pth')) # load finetuned weights
+
+tuned_model.eval()
+pretrained_model.eval()
+
+evaluator_finetuned = Evaluator(eval, train, tuned_model)
+print(evaluator_finetuned.evaluate())
+
+evaluator_pretrained = Evaluator(eval, train, pretrained_model)
+print(evaluator_pretrained.evaluate())
+```
+````
+
+The difference is shown in the tables below:
+
+| mAP@k  | pre-trained | fine-tuned |
+|--------|-------------|------------|
+| mAP@1  | 0.147       | 0.719      |
```

Comment:
how do you extract metrics at 1, 5, 10? The evaluator can't do that yet

Reply:
Using evaluator I get loop disconnect from AWS don't know why, that was from a custom script